### PR TITLE
Word embdedding example updated

### PR DESCRIPTION
**Pull request by 0bserver07**
*  The `Conv1D` class is missing from the layers, it looks like `Convolution1D` is used now.
* While using Python 3, loading data on line [64](https://github.com/fchollet/keras/blob/master/examples/pretrained_word_embeddings.py#L64) is failing. I found out that reading with ` encoding="latin-1"` fixes it.

### CODE REVIEW

**Comment by fchollet** [comment]
> The Conv1D class is missing from the layers, it looks like Convolution1D is used now.

It's aliased to `Conv1D` in the latest version. In general I would like to push people to use `Conv1D` /  `Conv2D` from now on, mostly to match the conventions set by TensorFlow.

> While using Python 3, loading data on line 64 is failing. I found out that reading with encoding="latin-1" fixes it.

Right. Then that should be made conditional on the Python version, e.g.: https://github.com/fchollet/keras/blob/master/keras/datasets/imdb.py#L110-L113

Note that this file uses `'` rather than `"` as the string delimiter.

**Comment by 0bserver07** [comment]
* Reverted this back. My bad, I thought the Conv1D is older, since I was looking at this part of the [documentation](https://keras.io/layers/convolutional/) that uses `keras.layers.convolutional.Convolution1D()` 

* Changed this to be conditional depending on versions. Also it's using `'` rather than `"` now 😄.

**Comment by ethen8181** [comment]
Hi, I have a quick question regarding this example: is there a reason why the word id mapping doesn't start from 0? If it does, then we won't have to initialize the embedding matrix with `len(word_index) + 1` and have an unused dimension 0.

```python
# numpy indices 0 not used?
embedding_matrix = np.zeros((len(word_index) + 1, EMBEDDING_DIM))
``` 

Or am I missing something? Thanks

**Comment by fchollet** [comment]
@ethen8181 that's done to reserve 0 as a padding character. This allows us to do sequence masking if needed.

**Comment by ethen8181** [comment]
@fchollet  Thanks for the response, and sorry to bother you, but I still have some other questions and a possible feature request. 

### Question

When we call the `fit_on_texts` method:

```python
tokenizer = Tokenizer(nb_words=MAX_NB_WORDS)
tokenizer.fit_on_texts(texts)
word_index = tokenizer.word_index
print('Found %s unique tokens.' % len(word_index))
```
Why is the total number of unique word_index still the number of unique tokens found in all of the documents instead of the `MAX_NB_WORDS` we've specified (if it is smaller than the number of unique tokens found)? 

And since all the less frequently appeared word will get removed when we pad them into sequences:
```python
X = pad_sequences( sequences, maxlen = MAX_SEQUENCE_LENGTH )
````
Why is the `embedding_matrix`'s row size the total number of unique of tokens found in all of the documents?
```python
embedding_matrix = np.zeros( ( len(word_index) + 1, EMBEDDING_DIM ), dtype = 'float32' )
```

### Feature Request

Remove english stop words.

```python
# just listing out the gist of it, a bunch of other code should be added
def text_to_word_sequence(text, filters=base_filter(), lower=True, split=" " ):
    '''prune: sequence of characters to filter out
    '''
    if lower:
        text = text.lower()
    text = text.translate(maketrans(filters, split*len(filters)))
    seq = text.split(split)
    # 1. no place to judge, but why call it _f?
    # 2. the ENGLISH_STOP_WORDS
    # https://gist.github.com/ethen8181/d57e762f81aa643744c2ffba5688d33a             
    return [ _f for _f in seq if _f and _f not in ENGLISH_STOP_WORDS ]             
```
